ddtrace/internal/datastreams/botocore.py

```diff
@@ -56,24 +58,67 @@ def inject_context(trace_data, endpoint_service, dsm_identifier):
     from . import data_streams_processor as processor
 
     path_type = "type:{}".format(endpoint_service)
+
+    payload_size = None
+    if endpoint_service == "sqs":
+        payload_size = calculate_sqs_payload_size(message, trace_data)
+    elif endpoint_service == "sns":
+        payload_size = calculate_sns_payload_size(message, trace_data)
+    elif endpoint_service == "kinesis":
+        payload_size = calculate_kinesis_payload_size(message, trace_data)
+
     if not dsm_identifier:
         log.debug("pathway being generated with unrecognized service: ", dsm_identifier)
-    ctx = processor().set_checkpoint(["direction:out", "topic:{}".format(dsm_identifier), path_type])
+    ctx = processor().set_checkpoint(
+        ["direction:out", "topic:{}".format(dsm_identifier), path_type], payload_size=payload_size
+    )
     DsmPathwayCodec.encode(ctx, trace_data)
 
 
-def handle_kinesis_produce(stream, dd_ctx_json):
+def calculate_sqs_payload_size(message, trace_data = None):
+    payload_size = _calculate_byte_size(message.get("MessageBody", ""))
+    payload_size += _calculate_byte_size(message.get("MessageAttributes", {}))
+    if trace_data:
+        # we should count datadog message attributes which aren't yet added to the message
+        payload_size += _calculate_byte_size({"_datadog": {"DataType": "String", "StringValue": trace_data}})
```

Comment:
nice!

Reply:
would it make sense to save&reuse the size of `{"_datadog": {"DataType": "String", "StringValue":}}`, and only compute the size of `trace_data` here ?
Or maybe it's not making much of a difference ?

(same remark for sns below)

Reply:
Not much of a difference IMO 🤷